stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -259,7 +259,7 @@ public class ChatClient internal constructor(
         }
         val userState = userStateService.state
         when {
-            userState is UserState.UserSet && userState.user.id == user.id && socketStateService.state == SocketState.Idle -> {
+            userState is UserState.UserSet && userState.user.id == user.id && (socketStateService.state == SocketState.Idle || socketStateService.state == SocketState.Disconnected) -> {
```

Comment:
For me it's not clear why we need to set user when the socket is disconnected.
We check for the idle socket state, because for pushes and background job we could set user without WS connection, and when we open app we want to set user with full flow (setting WS connection). For that this condition.
But if we were disconnected, we just released the WS connection not to waste resources. For this case, we don't need to reset user again

Reply:
When the activity is recreated we could come in this if the user chooses to initialises our SDK in the activity instead of the `Application` class. He/she will go into the else case... For me it makes more sense to connect again, because in the end it was a second request made by the activity. 

In this case removing the check for `SocketState.Disconnected` is less problematic, because there's no crash and we log what's going on... But asking for a connection in the `onCreate` of an activity doesn't seam too crazy for me, I think it makes to support it. What do you think?

Reply:
IMO, it's not appropriate usage of our SDK. UI lifecycle shouldn't impact set user flow @JcMinarro @adasiewiczr Guys, what do you think?

Reply:
``` if the user chooses to initialises our SDK in the activity instead of the Application class. He/she will go into the else case```
Are we setting `SocketState.Disconnected` during SDK initialization? I think it should be `disconnected` only if user was already set.
In our sample app, a user is connected in ViewModel and we don't have any problems with that.

Reply:
@adasiewiczr If you enable "Don't keep activities" observe that we always go to the else branch. But yes, the app continue working just fine

Reply:
Let's remove this than, no need to make this a super long discussion.